diff --git a/libretro/Makefile b/libretro/Makefile
index 3b564ae..853a231 100644
--- a/libretro/Makefile
+++ b/libretro/Makefile
@@ -150,7 +150,7 @@ else ifeq ($(platform), libnx)
    PORTLIBS := $(PORTLIBS_PATH)/switch
    TARGET := $(TARGET_NAME)_libretro_$(platform).a
    DEFINES := -DSWITCH=1 -D__SWITCH__ -DARM
-   CFLAGS := $(DEFINES) -fPIE -I$(PORTLIBS)/include/ -I$(LIBNX)/include/ -ffunction-sections -fdata-sections -ftls-model=local-exec -specs=$(LIBNX)/switch.specs
+   CFLAGS := $(DEFINES) -fPIC -I$(PORTLIBS)/include/ -I$(LIBNX)/include/ -ffunction-sections -fdata-sections -ftls-model=local-exec -specs=$(LIBNX)/switch.specs
    CFLAGS += -march=armv8-a -mtune=cortex-a57 -mtp=soft -mcpu=cortex-a57+crc+fp+simd -ffast-math
    CXXFLAGS := $(ASFLAGS) $(CFLAGS)
    STATIC_LINKING=1
